### PR TITLE
chore(startup): Break up `Application::prepare_from_opts` a bit

### PR DESCRIPTION
**Pull request by bruceg**
Basically just moving code around to make getting at the individual steps easier.

First stage of several to come.

### CODE REVIEW

**Review by tobz — APPROVED** [review]

File: src/app.rs [hunk 1]
```diff
@@ -455,3 +409,48 @@ async fn sources_finished(mutex: Arc<Mutex<TopologyController>>) {
         }
     }
 }
+
+fn get_log_levels(default: &str) -> String {
+    std::env::var("VECTOR_LOG")
+        .or_else(|_| {
+            warn!(message = "Use of $LOG is deprecated. Please use $VECTOR_LOG instead.");
+            std::env::var("LOG")
+        })
```
Comment (spencergilbert) [inline src/app.rs:418]:
```suggestion
        .or_else(|_| {
            warn!(message = "DEPRECATED: Use of $LOG is deprecated. Please use $VECTOR_LOG instead.");
            std::env::var("LOG")
        })
```

I know this was kept as-is but can we correct this message (per the deprecation docs). Also throwing a reminder into the void that we should remove this sooner than later, it's been a few releases.

File: src/app.rs [hunk 2]
```diff
@@ -455,3 +409,48 @@ async fn sources_finished(mutex: Arc<Mutex<TopologyController>>) {
...
+
+pub fn build_runtime(threads: Option<usize>, thread_name: &str) -> Result<Runtime, ExitCode> {
+    let mut rt_builder = runtime::Builder::new_multi_thread();
+    rt_builder.enable_all().thread_name(thread_name);
+
+    if let Some(threads) = threads {
+        if threads < 1 {
+            #[allow(clippy::print_stderr)]
+            {
+                eprintln!("The `threads` argument must be greater or equal to 1.");
```
Comment (spencergilbert) [inline src/app.rs:444]: `error!()` isn't available yet because the subscriber won't have been created yet?

**Review by spencergilbert — APPROVED** [review]
Mechanically things make sense, one random question and a request to clean up an incorrect deprecation log.

File: src/app.rs (hunk 1 above)
Comment (bruceg) [reply src/app.rs:418]: Will do. Looking at it, that code would also suggest we _always_ print that warning if `$VECTOR_LOG` is unset.

File: src/app.rs (hunk 2 above)
Comment (bruceg) [reply src/app.rs:444]: I _think_ so, and because we immediately exit, the error will go nowhere.

File: src/app.rs (hunk 1 above)
Comment (spencergilbert) [reply src/app.rs:418]: 🤔 I'd also think so.... but I don't think I see that log normally, and I'm usually not setting anything.

**Review by spencergilbert — APPROVED** [review]
Looks like you got caught on the intermittent component docs bug, hopefully passes on a retry.

**Pull request merged**